claims.go

```diff
@@ -63,6 +64,11 @@ func getClaim(logger log.Logger, allClaims map[string]interface{}, claim string)
 	//
 	// Or we fork and/or PR go-oidc.
 
+	switch v := val.(type) {
+	case float32:
+	case float64:
+		return strconv.Itoa(int(v))
```

Comment:
Should we wrap this in `json.Number()` so that it matches with the type that comes in from Vault's API? Similar to what's in the comment above.

Reply:
I don't think we can - the rest of the code expects the claim to be `string`, `bool` or `[]`: https://github.com/hashicorp/vault-plugin-auth-jwt/blob/main/claims.go#L125 That is why I leave it `string`.

Reply:
I tested this out using the Vault API with a numeric bound claims value and it doesn't work right now. The reason is that in `validateBoundClaims` the actual type becomes `string` and expected type is `json.Number`. This causes `matchFound` to return false.

This patch makes the two types match up as `json.Number`.

```
diff --git a/claims.go b/claims.go
index a032a29..d0fb028 100644
--- a/claims.go
+++ b/claims.go
@@ -4,6 +4,7 @@
 package jwtauth
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 	"strconv"
@@ -66,9 +67,9 @@ func getClaim(logger log.Logger, allClaims map[string]interface{}, claim string)
 
 	switch v := val.(type) {
 	case float32:
-		return strconv.Itoa(int(v))
+		return json.Number(strconv.Itoa(int(v)))
 	case float64:
-		return strconv.Itoa(int(v))
+		return json.Number(strconv.Itoa(int(v)))
 	}
 	return val
 }
@@ -184,7 +185,7 @@ func normalizeList(raw interface{}) ([]interface{}, bool) {
 	switch v := raw.(type) {
 	case []interface{}:
 		normalized = v
-	case string, bool:
+	case string, bool, json.Number:
 		normalized = []interface{}{v}
 	default:
 		return nil, false
```

I'd be in support of moving forward with this if you agree.

Reply:
OK sounds good to me! I just pushed a new commit to use `json.Number` and add the new type in the switch case.

Reply:
Thanks!